include/envoy/compressor/compressor.h

```diff
@@ -13,11 +13,11 @@ class Compressor {
   virtual ~Compressor() {}
 
   /**
-   * Compresses data from one buffer into another buffer.
-   * @param input_buffer supplies the buffer with data to be compressed.
-   * @param output_buffer supplies the buffer to output compressed data.
+   * Compresses data buffer.
+   * @param buffer supplies the reference to data to be compressed.
+   * @param trailer supplies the indication to write trailer at the end of compressed buffer.
```

Comment:
IMO "trailer" is a bit hard to understand. Personally I would just call it "finish" along with "supplies whether to finish the compression and flush any buffered compression data" WDYT?

Reply:
@mattklein123 @gsagula before I thought that having explicit APIs like

Reply:
@dio Go ahead if you think that having a more explicit API is easier to understand and reason about it. The above were just suggestions :)